.github/workflows/integrationtests.yaml

```diff
@@ -122,4 +126,54 @@ jobs:
       - name: Daps Integration Test
         env:
           RUN_INTEGRATION_TEST: true
-        run: ./gradlew clean extensions:iam:daps:check
\ No newline at end of file
+        run: ./gradlew clean extensions:iam:daps:check
+
+  Check-Cloud-Environments:
+    environment: Azure-dev
+    runs-on: ubuntu-latest
+    steps:
+      - id: has-azure
+        env:
+          HAS_AZURE: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        if: "${{ env.HAS_AZURE != '' }}"
+        run: echo "::set-output name=defined::true"
+    outputs:
+      has-azure: ${{ steps.has-azure.outputs.defined }}
+
+  Azure-Cloud-Integration-Test:
+    needs: [ Check-Cloud-Environments ]
+    if: needs.Check-Cloud-Environments.outputs.has-azure == 'true'
+    environment: Azure-dev
+    runs-on: ubuntu-latest
+
+    env:
+      RUN_INTEGRATION_TEST: true
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: 'Az CLI login'
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: "Fetch Terraform outputs"
+        run: printf "$TERRAFORM_OUTPUTS" > resources/azure/testing/terraform_outputs.json
+        env:
+          TERRAFORM_OUTPUTS: ${{ secrets.TERRAFORM_OUTPUTS }}
+
+      - name: "List Terraform output keys"
+        run: cat resources/azure/testing/terraform_outputs.json | jq -r 'keys | .[]'
```

Comment:
nit: why we need to list these keys ? just as a information or some other use here.

Reply:
Thanks, added comment. This is for manual info/debugging